pkg/compute/executor.go

```diff
@@ -56,26 +59,58 @@ func NewBaseExecutor(params BaseExecutorParams) *BaseExecutor {
 	}
 }
 
+func prepareInputVolumes(ctx context.Context, strgprovider storage.StorageProvider, inputSources ...*models.InputSource) ([]storage.PreparedStorage, func(context.Context) error, error) {
+	inputVolumes, err := storage.ParallelPrepareStorage(ctx, strgprovider, inputSources...)
+	if err != nil {
+		return nil, nil, err
+	}
+	return inputVolumes, func(ctx context.Context) error {
+		return storage.ParallelCleanStorage(ctx, strgprovider, inputVolumes)
+	}, nil
+}
+
+func prepareWasmVolumes(ctx context.Context, strgprovider storage.StorageProvider, wasmEngine wasmmodels.EngineSpec) (map[string][]storage.PreparedStorage, func(context.Context) error, error) {
+	importModuleVolumes, err := storage.ParallelPrepareStorage(ctx, strgprovider, wasmEngine.ImportModules...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	entryModuleVolumes, err := storage.ParallelPrepareStorage(ctx, strgprovider, wasmEngine.EntryModule)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	volumes := map[string][]storage.PreparedStorage{
+		"importModules": importModuleVolumes,
+		"entryModules":  entryModuleVolumes,
+	}
+
+	cleanup := func(ctx context.Context) error {
+		err1 := storage.ParallelCleanStorage(ctx, strgprovider, importModuleVolumes)
+		err2 := storage.ParallelCleanStorage(ctx, strgprovider, entryModuleVolumes)
+		if err1 != nil || err2 != nil {
+			return fmt.Errorf("Error cleaning up WASM volumes: %v, %v", err1, err2)
+		}
+		return nil
+	}
+
+	return volumes, cleanup, nil
+}
+
 func PrepareRunArguments(
 	ctx context.Context,
 	strgprovider storage.StorageProvider,
 	execution *models.Execution,
 	resultsDir string,
-	cleanup *system.CleanupManager,
-) (*executor.RunCommandRequest, error) {
-	inputVolumes, err := storage.ParallelPrepareStorage(ctx, strgprovider, execution.Job.Task().InputSources...)
+) (*executor.RunCommandRequest, func(context.Context) error, error) {
```

Comment:
Although `func(context.Context) error` is nice and explicit, do you think we should use a type to describe the Cleanup function as a little bit of extra inline documentation? Perhaps as 

```go
type CleanupFn(context.Context) error
```

Reply:
this has been addressed in a subsequent commit